ale/base/base.py

```diff
@@ -61,102 +48,249 @@ def to_dict(self):
         keys = set()
         return {p:getattr(self, p) for p in dir(self) if p[0] != "_" and isinstance(getattr(type(self), p), property)}
 
-
     @property
-    def label(self):
-        if not hasattr(self, "_label"):
-            if isinstance(self._file, pvl.PVLModule):
-                self._label = self._file
-            try:
-                self._label = pvl.loads(self._file)
-            except Exception:
-                self._label = pvl.load(self._file)
-            except:
-                raise ValueError("{} is not a valid label".format(self._file))
-        return self._label
+    @abc.abstractmethod
+    def image_lines(self):
+        """
+        Returns
+        -------
+        : int
+          Number of lines in image
+        """
+        pass
 
     @property
-    def file(self):
-        return self._file
+    @abc.abstractmethod
+    def image_samples(self):
+        """
+        Returns
+        -------
+        : int
+          Number of samples in image
+        """
+        pass
 
     @property
-    def interpolation_method(self):
-        return "lagrange"
+    @abc.abstractmethod
+    def usgscsm_distortion_model(self):
+        """
+        Returns
+        -------
+        : dict
+          A dict containing the information about the distortion model for the usgscsm
+        """
+        pass
+
+#   def is_valid(self):
```

Comment:
Should `is_valid` remain? It wasn't explicitly part of our design, so I commented it out.